src/findWithPagination.ts

```diff
@@ -30,24 +30,33 @@ export type FindOptions<T = any> = (FindManyOptions<T> | FindConditions<T>) & {
 export function findWithPagination<
   T extends ObjectLiteral,
   O extends FindOptions<T>
->(repository: Repository<T>, options: FindOptions<T>): Promise<Page<T>> {
+>(
+  repository: Repository<T>,
+  options: FindOptions<T>,
+  Paginator: typeof CursorPaginator = CursorPaginator)
```

Comment:
Introduce dependency injection for the dependency on the Paginator class

Reply:
Do we need this? Seems like YAGNI

Reply:
Just added it to enable smaller units of testing